### PR TITLE
Add fail-fast mode

### LINKED ISSUE #48: Flag to stop after first test failure
From https://github.com/sindresorhus/ava/issues/9#issuecomment-138639602.

Does it makes sense to have this?

`--bail`, `--fail-fast`, other?

-

On that note, any thoughts on failing-fast by default? Any reason you'd want the test to complete before failing? I'm sure there are since test runners usually run to completion. Just useful to have this documented either way.

### PR DESCRIPTION
**Pull request by vdemedes**
This PR adds `--fail-fast` option to exit after the first test failure.

Fixes #48.

Changes:
  - add `--fail-fast` option to `--help` output
  - add fail-fast mode
  - add tests

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -34,9 +35,15 @@ var cli = meow({
...
 	]
 }, {
-	string: ['_']
+	string: ['_'],
+	boolean: ['fail-fast']
 });
 
+// set env variable, so that forked process can detect the flag
+if (cli.flags.failFast) {
+	process.env.AVA_FAIL_FAST = true;
```
Comment (sindresorhus) [inline cli.js]: Would be cleaner to pass this as a command-line option when forking the process.

File: test/test.js [hunk 2]
```diff
@@ -756,3 +760,21 @@ test('display test title prefixes', function (t) {
...
+		t.ok(err);
+		t.is(err.code, 1);
+
+		var output = stderr
+			.replace(new RegExp(figures.tick, 'gm'), 'success')
+			.replace(new RegExp(figures.cross, 'gm'), 'error');
+
+		t.true(output.indexOf('error [anonymous] false fail false') >= 0);
+		t.false(output.indexOf('success [anonymous]') >= 0);
+		t.true(output.indexOf('1 test failed') >= 0);
```
Comment (sindresorhus) [inline test/test.js]: Use `!== -1` instead of `>= 0`

File: test/test.js [hunk 3]
```diff
@@ -756,3 +760,21 @@ test('display test title prefixes', function (t) {
...
+test('fail-fast mode', function (t) {
+	t.plan(5);
+
+	execCli(['fixture/fail-fast.js', '--fail-fast'], function (err, stdout, stderr) {
+		t.ok(err);
+		t.is(err.code, 1);
+
+		var output = stderr
+			.replace(new RegExp(figures.tick, 'gm'), 'success')
+			.replace(new RegExp(figures.cross, 'gm'), 'error');
```
Comment (sindresorhus) [inline test/test.js]: Instead of doing a replace here just put the `figures` directly into the `indexOf` check.

File: cli.js [hunk 4]
```diff
@@ -21,7 +21,8 @@ var cli = meow({
 		'  ava <file|folder|glob> [...]',
 		'',
 		'Options',
-		'  --init  Add AVA to your project',
+		'  --init       Add AVA to your project',
+		'  --fail-fast  Stop after first test failure',
```
Comment (sindresorhus) [inline cli.js]: Also needs to be added to the readme: https://github.com/sindresorhus/ava#cli

**Comment by vdemedes** [comment]
@sindresorhus Updates pushed

**Comment by sindresorhus** [comment]
Looks good :)

**Comment by vdemedes** [comment]
Thanks!

**Pull request merged**